witchcraft/witchcraft.go

```diff
@@ -615,10 +616,20 @@ func (s *Server) Start() (rErr error) {
 	}
 	internalHealthCheckSources := []healthstatus.HealthCheckSource{configReloadHealthCheckSource}
 
-	// extract the TCP JSON receiver from the runtime config
-	servicesCfg := baseRefreshableRuntimeCfg.CurrentBaseRuntimeConfig().ServiceDiscovery
-	receiverCfg, err := servicesCfg.MustClientConfig("sls-log-tcp-json-receiver")
-	if err == nil && len(receiverCfg.URIs) > 0 {
+	// enable TCP logging if the envelope metadata and the TCP receiver are both configured
+	receiverCfg := baseRefreshableRuntimeCfg.CurrentBaseRuntimeConfig().ServiceDiscovery.ClientConfig("sls-log-tcp-json-receiver")
+	envelopeMetadata, err := tcpjson.GetEnvelopeMetadata()
+	if err != nil {
+		if len(receiverCfg.URIs) > 0 {
```

Comment:
Can you add a comment after this block to explain why it's okay to proceed without warning/logging?